### PR TITLE
[SPARK-30834][DOCS][PYTHON][2.4] Add note for recommended pandas and pyarrow versions

### PR DESCRIPTION
**Pull request by BryanCutler**
### What changes were proposed in this pull request?

Add doc for recommended pandas and pyarrow versions. 

### Why are the changes needed?

The recommended versions are those that have been thoroughly tested by Spark CI. Other versions may be used at the discretion of the user.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

NA

### CODE REVIEW

**Comment by BryanCutler** [comment]
cc @HyukjinKwon @dongjoon-hyun

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you for the clarification, @BryanCutler .
Since @HyukjinKwon is already in the loop, cc @holdenk , too.

File: docs/sql-pyspark-pandas-with-arrow.md [hunk 1]
```diff
@@ -166,6 +166,12 @@ different than a Pandas timestamp. It is recommended to use Pandas time series f
 working with timestamps in `pandas_udf`s to get the best performance, see
 [here](https://pandas.pydata.org/pandas-docs/stable/timeseries.html) for details.
 
+### Recommended Pandas and PyArrow Versions
+
+For usage with pyspark.sql, the supported versions of Pandas is 0.19.2 and PyArrow is 0.8.0. Higher
```
Comment (dongjoon-hyun) [inline docs/sql-pyspark-pandas-with-arrow.md]: BTW, I forgot the exact versions of the `branch-2.4` PR builders in AmpLab Jenkins. These are the community testing version. Did I understand correctly?

**Review by zero323 — APPROVED** [review]

File: docs/sql-pyspark-pandas-with-arrow.md (hunk 1 above)
Comment (HyukjinKwon) [reply docs/sql-pyspark-pandas-with-arrow.md]: Yeah, i believe so.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to branch-2.4.

I addressed my own minor comment and verified the doc build. PR builders don't run doc builds.

**Comment by BryanCutler** [comment]
Thanks @HyukjinKwon !

**Pull request closed without merging**